classes/core/PKPRoutingProvider.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace PKP\core;
+
+use PKP\core\PKPContainer;
+use Illuminate\Http\Request;
+use PKP\middlewares\HasUser;
+use PKP\middlewares\HasRoles;
+use Illuminate\Routing\Router;
+use PKP\middlewares\HasContext;
+use Illuminate\Routing\UrlGenerator;
+use PKP\middlewares\AllowCrossOrigin;
+use PKP\middlewares\ValidateCsrfToken;
+use PKP\middlewares\RemoveTrailingSlash;
+use Psr\Http\Message\ServerRequestInterface;
+use Illuminate\Routing\RoutingServiceProvider;
+use PKP\middlewares\DecodeApiTokenWithValidation;
+use PKP\middlewares\SetupContextBasedOnRequestUrl;
+use Illuminate\Foundation\Http\Middleware\TrimStrings;
+use Illuminate\Foundation\Http\Middleware\ValidatePostSize;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull;
+use Illuminate\Contracts\Routing\UrlGenerator as UrlGeneratorContract;
+
+class PKPRoutingProvider extends RoutingServiceProvider
+{
+    protected static $globalMiddlewares = [
+        AllowCrossOrigin::class,
+        SetupContextBasedOnRequestUrl::class,
+        DecodeApiTokenWithValidation::class,
+        ValidateCsrfToken::class,
+        RemoveTrailingSlash::class,
+        ValidatePostSize::class,
+        TrimStrings::class,
+        ConvertEmptyStringsToNull::class,
+    ];
+
+    /**
+     * The application's route middleware.
+     * These middleware can/should be assigned to specific routes or routes groups individually.
+     */
+    protected array $routeMiddlewares = [
```

Comment:
This list is going to grow out of control pretty quickly when we start making broad use of the new routing toolset, I think - especially if we eventually unify our authorization classes with Laravel. There are lots of those and they are complex!

Would it be possible to delegate building this list to the individual controller, rather than centralizing it here?

Reply:
This is just to maintain a list of middleware that may use a lot and given those an alias name. we can always apply any new middleware on each of routes as we like 
```php
Route::put('/post/{id}', function (string $id) {
    // ...
})->middleware('role:editor');

// or

Route::put('/post/{id}', function (string $id) {
    // ...
})->middleware([SomeMiddleware::class, ...]);
```

Reply:
Ah, that makes sense!